### PR TITLE
Add back compression and encryption 

### PR DESCRIPTION
**Pull request by lynnliu030**

### CODE REVIEW

File: skyplane/api/dataplane.py [hunk 1]
```diff
@@ -237,6 +237,7 @@ def copy_log(instance):
             instance.download_file("/tmp/gateway.stdout", out_file)
             instance.download_file("/tmp/gateway.stderr", err_file)
 
+        print("COPY GATEWAY LOGS")
```
Comment (sarahwooders) [inline skyplane/api/dataplane.py:240]: Could you remove this?

File: skyplane/api/dataplane.py [hunk 2]
```diff
@@ -309,6 +310,7 @@ def run_async(self, jobs: List[TransferJob], hooks: Optional[TransferHook] = Non
         """
         if not self.provisioned:
             logger.error("Dataplane must be pre-provisioned. Call dataplane.provision() before starting a transfer")
+        print("discord", jobs)
```
Comment (sarahwooders) [inline skyplane/api/dataplane.py:313]: remove

File: skyplane/planner/planner.py [hunk 3]
```diff
@@ -240,13 +240,19 @@ def plan(self, jobs: List[TransferJob]) -> TopologyPlan:
             mux_or = src_program.add_operator(GatewayMuxOr(), parent_handle=obj_store_read, partition_id=partition_id)
             for i in range(n_instances):
                 src_program.add_operator(
-                    GatewaySend(target_gateway_id=dst_gateways[i].gateway_id, region=src_region_tag, num_connections=self.n_connections),
+                    GatewaySend(
```
Comment (sarahwooders) [inline skyplane/planner/planner.py:243]: Could you modify all the `GatewaySend` and `GatewayReceive` operators in the other planners to also include compression and encryption?

**Comment by sarahwooders** [comment]
So in the current implementation, my understanding is that each time data moves between a VM it is compressed and decompressed? This is fine for now, but maybe we should add an issue so in the future the data is compressed/encrypted just once when it is read from the object store.

**Review by sarahwooders — APPROVED** [review]
Added a few minor nits but looks great otherwise! Feel free to merge once cleaned up.

**Comment by lynnliu030** [comment]
> So in the current implementation, my understanding is that each time data moves between a VM it is compressed and decompressed? This is fine for now, but maybe we should add an issue so in the future the data is compressed/encrypted just once when it is read from the object store.

@sarahwooders I don't think so? all of these are specified in the gateway program; it'll only be decompressed or decrypted if you set it explicitly in the gateway programs of VMs located in destination regions

**Pull request merged**